subsys/llext/llext_load.c

```diff
@@ -542,6 +542,32 @@ static int llext_copy_symbols(struct llext_loader *ldr, struct llext *ext,
 	return 0;
 }
 
+/*
+ * Free resources only used during loading. Also, in case an error was
+ * detected, free everything that was allocated.
+ */
+static void llext_load_cleanup(struct llext_loader *ldr, struct llext *ext, int ret)
+{
+	/* free resources only used during loading */
+	llext_free(ldr->sect_map);
+	ldr->sect_map = NULL;
+
+	if (!IS_ENABLED(CONFIG_LLEXT_LOG_LEVEL_DBG)) {
+		/* keep the computed symbol tables for debugging */
```

Comment:
what if this is called in an error case? Maybe swap around as
```
	if (ret != 0) {
		/* free sections and exported symbols */
		...
	} else if (IS_ENABLED(CONFIG_LLEXT_LOG_LEVEL_DBG)) {
		return;
	}
	/* free the symbol table */
	...
```

Reply:
Indeed, thanks for spotting this! I will refactor this appropriately.

Reply:
In error case, `ext->exp_tab.syms` is "free'ed twice", but is already NULL on second `free` so nothing bad happens (`free(NULL)` is no-op in Zephyr).
I agree with your comment nonetheless (and made the same one a few lines below :smile:)